### PR TITLE
Observability 2.4.0

### PR DESCRIPTION
**Pull request by ps48**
### Description
Add observability 2.4.0 to manifest

### Issues Resolved
Resolves: https://github.com/opensearch-project/observability/issues/869

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by bbarani — APPROVED** [review]

File: manifests/2.4.0/opensearch-2.4.0.yml [hunk 1]
```diff
@@ -92,3 +92,12 @@ components:
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
+  - name: opensearch-observability
+    repository: https://github.com/opensearch-project/observability.git
+    ref: '2.x'
+    platforms:
+      - linux
```
Comment (bbarani) [inline manifests/2.4.0/opensearch-2.4.0.yml:99]: Need to add Windows to 2.4 builds

File: manifests/2.4.0/opensearch-dashboards-2.4.0.yml [hunk 2]
```diff
@@ -14,6 +14,10 @@ components:
     repository: https://github.com/opensearch-project/notifications.git
     working_directory: dashboards-notifications
     ref: '2.x'
+  - name: observabilityDashboards
+    repository: https://github.com/opensearch-project/observability.git
+    working_directory: dashboards-observability
+    ref: '2.x'
```
Comment (bbarani) [inline manifests/2.4.0/opensearch-dashboards-2.4.0.yml:20]: Need to add Windows to all 2.4 builds

Comment (bbarani) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:20]: @peterzhuamazon Can we add Windows target ourself for 2.4.0 release?

File: manifests/2.4.0/opensearch-2.4.0.yml (hunk 1 above)
Comment (ps48) [reply manifests/2.4.0/opensearch-2.4.0.yml:99]: Added windows to opensearch-2.4.0.yml

File: manifests/2.4.0/opensearch-dashboards-2.4.0.yml (hunk 2 above)
Comment (ps48) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:20]: @bbarani Should I go ahead platform fields in OpenSearch Dahsboards? I don't see it in any other plugin.

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-dashboards-2.4.0.yml:20]: @ps48 ignore dashboards for now as I am still testing windows with dashboards.
Thanks.

**Comment by peterzhuamazon** [comment]
ignore groovy test failing it is not related.

**Pull request merged**